test/e2e/query_test.go

```diff
@@ -368,31 +368,35 @@ func checkNetworkRequests(t *testing.T, addr string) {
 	ctx, cancel := chromedp.NewContext(context.Background())
 	t.Cleanup(cancel)
 
-	var networkErrors []string
+	testutil.Ok(t, runutil.Retry(1*time.Minute, ctx.Done(), func() error {
+		var networkErrors []string
 
-	// Listen for failed network requests and push them to an array.
-	chromedp.ListenTarget(ctx, func(ev interface{}) {
-		switch ev := ev.(type) {
-		case *network.EventLoadingFailed:
-			networkErrors = append(networkErrors, ev.ErrorText)
-		}
-	})
-
-	err := chromedp.Run(ctx,
-		network.Enable(),
-		chromedp.Navigate(addr),
-		chromedp.WaitVisible(`body`),
-	)
-	testutil.Ok(t, err)
+		err := chromedp.Run(ctx,
+			network.Enable(),
+			chromedp.Navigate(addr),
+			chromedp.WaitVisible(`body`),
+		)
 
-	err = func() error {
-		if len(networkErrors) > 0 {
-			return fmt.Errorf("some network requests failed: %s", strings.Join(networkErrors, "; "))
+		if err != nil {
+			return err
 		}
-		return nil
-	}()
 
-	testutil.Ok(t, err)
+		// Listen for failed network requests and push them to an array.
+		chromedp.ListenTarget(ctx, func(ev interface{}) {
```

Comment:
Shouldn't this happen before line 371? :thinking: Now we are doing this *after* the requests at the first iteration.

Reply:
Thanks for the review!
Do you mean outside the retry function? 
Well as we can see in the error logs("some network request failed") it fails because the size of networkErrors array is >0 and if we call the ListenTarget outside, will it append errors to this array in all iterations.

Reply:
My suggestion was to clean the slice on each iteration. I suppose the current design is OK as you say but don't we need to register this callback (https://pkg.go.dev/github.com/chromedp/chromedp#ListenTarget) before calling `chromedp.Run()`?

Reply:
You are right we should. Done :+1: